PCDHA12.py

```diff
@@ -0,0 +1,121 @@
+import os
+from abc import ABC, abstractmethod
+
+import numpy as np
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.SeqUtils import GC
+
+
+# 2.4
+def filter_length(record: SeqRecord, length_bounds: tuple) -> bool:
+    return length_bounds[0] <= len(record.seq) <= length_bounds[1]
+
+
+def filter_gc(record: SeqRecord, gc_bounds: tuple) -> bool:
+    return gc_bounds[0] <= GC(record.seq) <= gc_bounds[1]
+
+
+def filter_quality(record: SeqRecord, quality_threshold: int) -> bool:
+    avg_quality = np.mean(record.letter_annotations["phred_quality"])
+    return avg_quality >= quality_threshold
+
+
+def filter_fastq(input_path: str,
+                 gc_bounds: tuple = (0, 100),
+                 length_bounds: tuple = (0, 2 ** 32),
+                 quality_threshold: int = 0) -> None:
+    """ Reads a FASTQ file, filters sequences based on GC content, sequence
+    length and quality threshold, and writes the filtered sequences to
+    a new file """
```

Comment:
Спасибо за докстринг! Но мне казалось, что туда необходимо добавлять, какой инпут и какой аутпут функции идет. Что-то вот такое
```suggestion
    """
     Reads a FASTQ file, filters sequences based on GC content, sequence
    length and quality threshold, and writes the filtered sequences to
    a new file
        Args:
        input_path (str): Path to the input FASTQ file.
        gc_bounds (tuple): A tuple of two integers specifying the lower and
            upper bounds of the GC content, in percent. Sequences with GC
            content outside these bounds will be filtered out.
        length_bounds (tuple): A tuple of two integers specifying the lower and
            upper bounds of the sequence length, in base pairs. Sequences
            with length outside these bounds will be filtered out.
        quality_threshold (int): An integer specifying the minimum average
            quality score for a sequence to be kept. Sequences with average
            quality score below this threshold will be filtered out.
      """
```